pbs_light.go

```diff
@@ -406,6 +409,44 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	mRequestTimer.UpdateSince(pbs_req.Start)
 }
 
+// checkForValidBidSize goes through list of bids & find those which are banner mediaType and with height or width not defined
+// determine the num of ad unit sizes that were used in corresponding bid request
+// if num_adunit_sizes == 1, assign the height and/or width to bid's height/width
+// if num_adunit_sizes > 1, reject the bid (remove from list) and return an error
+// return updated bid list object for next steps in auction
+func checkForValidBidSize(bids pbs.PBSBidSlice, bidder *pbs.PBSBidder) pbs.PBSBidSlice {
+	var finalValidBids pbs.PBSBidSlice
+	for _, bid := range bids {
+		if bid.CreativeMediaType == "banner" && (bid.Height == 0 || bid.Width == 0) {
+			for _, adunit := range bidder.AdUnits {
+				var bidSizes []openrtb.Format
+				if adunit.BidID == bid.BidID {
+					for _, tmpSizes := range adunit.Sizes {
+						bidSizes = append(bidSizes, tmpSizes)
+					}
+				}
+
+				if len(bidSizes) == 1 {
+					bid.Width, bid.Height = bidSizes[0].W, bidSizes[0].H
+					finalValidBids = append(finalValidBids, bid)
+				} else if len(bidSizes) > 1 {
+					msg := `Detected WxH sizes in bid response were undefined for a bid request utilizing multiple adunit sizes.  
```

Comment:
message doesn't have to be this verbose.`Ignoring bid because no size defined` + biddercode would work.